### ISSUE TITLE
Bug: No nullish coalescing assignment operator

### ISSUE THREAD

**Issue by joakim-ronning** [eslint/eslint#17036]
### Environment

Node version: `16.13.0`
npm version: `9.5.0`
Local ESLint version: `^8.36.0`
Global ESLint version: `None`
Operating System: `MacOS Ventura 13.2.1`

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "extends": [
    "airbnb-base",
    "prettier"
  ],
  "env": {
    "node": true,
    "jest": true
  },
  "globals": {
    "jest": true,
    "expect": true
  },
  "plugins": [
    "prettier"
  ],
  "rules": {
     "no-underscore-dangle" : 0,
     "prettier/prettier": "error"
  },
  "parserOptions": {
    "ecmaVersion": 2020
  }
}

```
</details>

```js
  let x;
  x ??= 1;
```

### What did you expect to happen?

Nullish coalescing assignment should be recognized by ESLint.

### What actually happened?

results in a `Parsing error: Unexpected token =`

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @joakim-ronning, thanks for the issue!

Logical assignment operators were added in ES2021, so it is expected that parsing with `"ecmaVersion": 2020` results in an error. You'll need to update `ecmaVersion` in your config file:

```diff
"parserOptions": {
-    "ecmaVersion": 2020
+    "ecmaVersion": 2021
}
```
